sos/plugins/sas3ircu.py

```diff
@@ -0,0 +1,47 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+
+# This sosreport plugin is meant for sas adapters.
+# This plugin logs inforamtion on each adapter it finds.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+import os
+
+
+class SAS3ircu(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+
+    plugin_name = "sas3ircu"
+
+    def setup(self):
+
+        # get list of adapters
+        output = self.call_ext_prog("sh -c 'sas3ircu list'", timeout=5)
+
+        if not (output["status"]):  # status = 0 means no timeout
```

Comment:
Please compare `output["status"] == 0` (as you will find in other plugins). This avoids unpredictable behaviour if for any reason the status is `NoneType`.